### PR TITLE
[BACKLOG-35959] Add drag drop capabilities to File Browser.

### PR DESCRIPTION
**Pull request by tkafalas**

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 41m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 4 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.fromByteArray</b> (click to expand)</summary><p>

```
Could not load SWT library. Reasons: 
 no swt-win32-4623 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 no swt-win32 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32-4623.so
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32.so

```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeIds</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeNames</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.javaToNativeAndBack</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

  
Tests run: **4083**, Failures: **4**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 30m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeIds</b> (click to expand)</summary><p>

```
Could not load SWT library. Reasons: 
 no swt-win32-4623 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 no swt-win32 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32-4623.so
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32.so

```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeNames</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.javaToNativeAndBack</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

  
Tests run: **4081**, Failures: **3**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 28m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeIds</b> (click to expand)</summary><p>

```
Could not load SWT library. Reasons: 
 no swt-win32-4623 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 no swt-win32 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32-4623.so
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-win32.so

```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeNames</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.javaToNativeAndBack</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

  
Tests run: **4081**, Failures: **3**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 37m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeIds</b> (click to expand)</summary><p>

```
Could not load SWT library. Reasons: 
 no swt-pi-gtk-4623 in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 no swt-pi-gtk in java.library.path: [/usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
 /home/buildguy/.swt/lib/linux/x86_64/libswt-pi-gtk-4623.so: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory
 Can't load library: /home/buildguy/.swt/lib/linux/x86_64/libswt-pi-gtk.so

```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.getTypeNames</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

<details><summary><b>org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransferTest.javaToNativeAndBack</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.di.plugins.fileopensave.dragdrop.ElementTransfer
```

</p></details>

  
Tests run: **5316**, Failures: **3**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 54s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 127
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 20s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 127
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 20s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **4081**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/11/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **4081**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/12/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: plugins/file-open-save-new/api/src/test/java/org/pentaho/di/plugins/fileopensave/api/overwrite/OverwriteStatusTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
+package org.pentaho.di.plugins.fileopensave.api.overwrite;
```
[the remark this answers is not in the document]
Comment (tkafalas) [reply plugins/file-open-save-new/api/src/test/java/org/pentaho/di/plugins/fileopensave/api/overwrite/OverwriteStatusTest.java:1]: Noted, will fix

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/file-open-save-new/api,plugins/file-open-save-new/core,plugins/pur/core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **4081**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8888/13/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by ddiroma — APPROVED** [review]

**Pull request merged**